app/main.py

```diff
@@ -76,3 +102,131 @@ def get_share_link():
+        "https://www.googleapis.com/auth/drive.file", "https://www.googleapis.com/auth/drive"
+    ]
+    credential = ServiceAccountCredentials.from_json_keyfile_dict(
+        SPREADSHEET_CREDENTIALS, scope)
+    client = gspread.authorize(credential)
+    gsheet = client.open("organ_sheets").sheet1
+    data = gsheet.get_all_records()
+    return jsonify(data)
+
+
+@app.route("/program", methods=['GET', 'POST'])
+# Get the program information from Gen3 Data Commons
+def program():
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/', headers=HEADER)
+
+    json_data = json.loads(res.content)
+    program_list = []
+    for ele in json_data['links']:
+        program_list.append(ele.replace(
+            "/v0/submission/", ""))
+    new_json_data = {'program': program_list}
+    return new_json_data
+
+
+@app.route("/<program>/project", methods=['GET'])
+# Get all projects information from Gen3 Data Commons
+def project(program):
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/{program}', headers=HEADER)
+
+    json_data = json.loads(res.content)
+    project_list = []
+    for ele in json_data['links']:
+        project_list.append(ele.replace(
+            f"/v0/submission/{program}/", ""))
+    new_json_data = {'project': project_list}
+    return new_json_data
+
+
+@app.route("/dictionary", methods=['GET', 'POST'])
+# Get all dictionary node from Gen3 Data Commons
+def dictionary():
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/_dictionary', headers=HEADER)
+
+    json_data = json.loads(res.content)
+    dictionary_list = []
+    for ele in json_data['links'][2:30]:
+        dictionary_list.append(ele.replace(
+            "/v0/submission/_dictionary/", ""))
+        # dictionary_list.append(ele.replace(
+        #     "/v0/submission/_dictionary/", "").replace("_", " ").title())
+    new_json_data = {"dictionary": dictionary_list}
+    return new_json_data
+
+
+@app.route('/nodes/<node_type>', methods=['GET', 'POST'])
+# Exports all records in a dictionary node
+def export_node(node_type):
+    post_data = request.get_json()
+    program = post_data.get('program')
+    project = post_data.get('project')
+    format = post_data.get('format')
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/{program}/{project}/export/?node_label={node_type}&format={format}', headers=HEADER)
+    return res.content
+
+
+@app.route('/records/<uuid>', methods=['GET', 'POST'])
+# Exports one or more records, use comma to separate the uuids
+# e.g. uuid1,uuid2,uuid3
+def export_record(uuid):
+    post_data = request.get_json()
+    program = post_data.get('program')
+    project = post_data.get('project')
+    format = post_data.get('format')
+    res = requests.get(
+        f'{Gen3Config.GEN3_ENDPOINT_URL}/api/v0/submission/{program}/{project}/export/?ids={uuid}&format={format}', headers=HEADER)
```

Comment:
Would this work with GET method with the program, project and format being undefined?